sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -106,22 +106,43 @@ private[sql] class RocksDBStateStoreProvider
     override def metrics: StateStoreMetrics = {
       val rocksDBMetrics = rocksDB.metrics
       def commitLatencyMs(typ: String): Long = rocksDBMetrics.lastCommitLatencyMs.getOrElse(typ, 0L)
-      def avgNativeOpsLatencyMs(typ: String): Long = {
-        rocksDBMetrics.nativeOpsLatencyMicros.get(typ).map(_.avg).getOrElse(0.0).toLong
+      def nativeOpsLatencyMillis(typ: String): Long = {
+        rocksDBMetrics.nativeOpsMetrics.get(typ).map(_ * 1000).getOrElse(0)
+      }
+      def sumNativeOpsLatencyMillis(typ: String): Long = {
+        rocksDBMetrics.nativeOpsHistograms.get(typ).map(_.sum / 1000).getOrElse(0)
+      }
+      def nativeOpsCount(typ: String): Long = {
+        rocksDBMetrics.nativeOpsHistograms.get(typ).map(_.count).getOrElse(0)
+      }
```

Comment:
`nativeOpsCount` looks like more for `nativeOpsMetrics`?

Reply:
`nativeOpsCount` is extracting the `count` from histogram based metrics. `nativeOpsMetrics` is extracting the value from ticker based metrics.

Reply:
`nativeOpsCount` is extracting the `count` from histogram based metrics. `nativeOpsMetrics` is extracting the value from ticker based metrics.

Reply:
Oh, I see. Thanks for clarifying it.